packages/mongo/collection.js

```diff
@@ -437,6 +432,77 @@ var throwIfSelectorIsNotId = function (selector, methodName) {
  * @param {Object} doc The document to insert. May not yet have an _id attribute, in which case Meteor will generate one for you.
  * @param {Function} [callback] Optional.  If present, called with an error object as the first argument and, if no error, the _id as the second.
  */
+Mongo.Collection.prototype.insert = function insert(doc, callback) {
+  // Make sure we were passed a document to insert
+  if (!doc) {
+    throw new Error("insert requires an argument");
+  }
+
+  // Shallow-copy the document and possibly generate an ID
+  doc = _.extend({}, doc);
+
+  if ('_id' in doc) {
+    if (!doc._id || !(typeof doc._id === 'string'
+          || doc._id instanceof Mongo.ObjectID)) {
+      throw new Error("Meteor requires document _id fields to be non-empty strings or ObjectIDs");
+    }
+  } else {
+    let generateId = true;
+
+    // Don't generate the id if we're the client and the 'outermost' call
+    // This optimization saves us passing both the randomSeed and the id
+    // Passing both is redundant.
+    if (this._isRemoteCollection()) {
+      const enclosing = DDP._CurrentInvocation.get();
+      if (!enclosing) {
+        generateId = false;
+      }
+    }
+
+    if (generateId) {
+      doc._id = this._makeNewID();
+    }
+  }
+
+  // On inserts, always return the id that we generated; on all other
+  // operations, just return the result from the collection.
+  var chooseReturnValueFromCollectionResult = function (result) {
+    if (doc._id) {
+      return doc._id;
+    }
+
+    // XXX what is this for??
```

Comment:
What came out of this, I thought we talked about this and you figured out some way around it?